compiler/il/OMRNode.cpp

```diff
@@ -2443,6 +2493,18 @@ OMR::Node::computeIsCollectedReference()
          continue;
          }
 
+      if (op.isTernary())
+         {
+         TR_YesNoMaybe secondChildResult = curNode->getSecondChild()->computeIsCollectedReferenceImpl(processedNodesCollected, processedNodesNotCollected);
+         if (TR_maybe == secondChildResult)
+            {
+            TR_YesNoMaybe thirdChildResult = curNode->getThirdChild()->computeIsCollectedReferenceImpl(processedNodesCollected, processedNodesNotCollected);
+            return recordProcessedNodeResult(receiverNode, thirdChildResult, processedNodesCollected, processedNodesNotCollected);
+            }
+         else
+            return recordProcessedNodeResult(receiverNode, secondChildResult, processedNodesCollected, processedNodesNotCollected);
+         }
+
       // opcodes associated with a symref, we should
       // be able to tell its collectedness from its symref.
       if (op.isLoadVar() || op.isLoadAddr() || op.isLoadReg())
```

Comment:
I worry when I see a register load's symbol reference being accessed since technically I don't think a register load opcode returns true for hasSymbolReference. I know there may be other places in the codebase that may be doing something similar but I want to ask if the register load evaluator figures out if the value is collected or not via some other logic and if so, if that is what you could be using here.

Reply:
I checked the regLoad evaluator, it also assumes and queries the symbol to check for collected reference. So the logic seems consistent.